api/store/database.go

```diff
@@ -10,9 +10,12 @@ import (
 type Database interface {
 	PutUser(*user.User) error
 	GetUser(string) (*user.User, error)
+	UpdateUser(*user.User) error
 
 	PutProject(*project.Project) error
+	ProjectExists(string) bool
 	GetProject(string) (*project.Project, error)
```

Comment:
use this one with name

Reply:
as in ProjectNameExists?

Reply:
sorry... use GetProject(string) as GetProjectByName(string) or get rid of it

Reply:
I think it should be left, its a boolean doesn't seem right to pass an error for no reason

Reply:
hmm but I guess i return an error anyway when thats called...

Reply:
but it still seems weird to me that the server would fetch someone elses project from the database if the name exists (even if doesnt get returned to the client)

Reply:
just return a 403 if its not in the user object and say forbidden even if it doesnt exist

Reply:
i didnt mean change the boolean func. I think you should get rid of "ProjectNameExists"... I was talking about having both `GetProject(string) (*project.Project, error)` and `GetProjectByName(string) (*project.Project, error)`. The old one [pressumably used Id, which we dont have anymore, so we shouldnt use it.

Reply:
it will never be in the user project because this is called in create project

Since we're using global names this is to check if a name has been taken when a user creates a new project. I think from a readability point it makes sense to leave this, doesnt seem right to try fetch a project when creating one

Reply:
I think we didnt understand each other. what i was saying is kill GetProject as below